### PR TITLE
add support for HACBS service provider workspace

### PR DESCRIPTION
**Pull request by MatousJobanek**
* refactoring of `hack/configure-kcp.sh` to use functions so we can configure appstudio & hacbs worskpace via one function
* changed the templating of `argo-cd-apps/base/authorization-in-kcp.yaml` to use matrix so it's applied for both SP workspaces
* slightly changed the structure of `components/authorization/kcp` by adding:
    * base - which is applied for both workspaces
    * appstudio - only for appstudio specific authz
    * hacbs - only for HACBS specific authz

with these changes, we will be able to create ApplicationSet using `redhat-hacbs-workspace-{{kcp-name}}` as the destination name for all HACBS components

!!! DO NOT MERGE YET !!! waiting until the `redhat-hacbs` service provider workspace is created in CPS

cc @ralphbean

### CODE REVIEW

File: hack/configure-kcp.sh [hunk 1]
```diff
@@ -21,48 +21,44 @@ function extra_params() {
...
+    if grep -q "insecure-skip-tls-verify: true" ${KCP_KUBECONFIG}; then
+      sed -i 's/certificate-authority-data: .*/insecure-skip-tls-verify: true/' /tmp/${SYNC_TARGET}-syncer.yaml
+    fi
+    kubectl apply -f /tmp/${SYNC_TARGET}-syncer.yaml --kubeconfig ${CLUSTER_KUBECONFIG}
+  fi
+  
+  BIND_SCOPE="system:authenticated"
+  if [[ ${KCP_INSTANCE_NAME} == "kcp-stable" ]] || [[ ${KCP_INSTANCE_NAME} == "kcp-unstable" ]] && [[ ${ROOT_WORKSPACE} == "root" ]]
+  then
+    BIND_SCOPE="rh-sso:16270929"
```
Comment (scoheb) [inline hack/configure-kcp.sh:57]: can we leave a comment here as to what this refers to?

Comment (scoheb) [reply hack/configure-kcp.sh:57]: actually, I now see this was already here...nonetheless if you know, a comment would be nice

**Comment by ralphbean** [comment]
Wanted to verify preview and bootstrap changes in kcp-stable against my user workspace, but it is having some OOM issues today: https://coreos.slack.com/archives/C035ZBD755E/p1663262918662849?thread_ts=1663262799.902799&cid=C035ZBD755E

**Comment by ralphbean** [comment]
OK, with kcp-stable working again, I can confirm bootstrap in preview mode against kcp-stable works as well as expected.

**Comment by ralphbean** [comment]
@MatousJobanek I think this needs an additional change in `hack/destroy.sh` to add removal of the redhat-hacbs workspace.

File: hack/configure-kcp.sh (hunk 1 above)
Comment (MatousJobanek) [reply hack/configure-kcp.sh:60]: added comment in 966caac
I hope that it clarifies the purpose of it

**Comment by MatousJobanek** [comment]
@ralphbean Good catch - I totally forgot to update that as well. 
Done in [d358b12](https://github.com/redhat-appstudio/infra-deployments/pull/729/commits/d358b1263d9d4b4c9175d4b154250b16982cd2fe)
Thanks

**Review by ralphbean — APPROVED** [review]

**Comment by ralphbean** [comment]
Awaiting https://coreos.slack.com/archives/C035ZBD755E/p1662729348114149

**Review by Michkov — APPROVED** [review]

**Comment by ralphbean** [comment]
OK, I think https://issues.redhat.com/browse/KCP-228 is done now and this can be merged.

**Pull request merged**